sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -251,7 +253,11 @@ class Dataset[T] private[sql](
           case seq: Seq[_] => seq.mkString("[", ", ", "]")
           case _ => cell.toString
         }
-        if (truncate && str.length > 20) str.substring(0, 17) + "..." else str
+        if (truncateTo > 0 && str.length > truncateTo) {
+          // do not show ellipses for strings shorter than 4 characters.
+          if (truncateTo < 4) str.substring(0, truncateTo)
+          else str.substring(0, truncateTo - 3) + "..."
```

Comment:
I did not like this check, but added it to ensure we do not deviate from the existing default behavior.